cookbooks/aws-parallelcluster-environment/test/controls/cloudwatch_spec.rb

```diff
@@ -14,11 +14,18 @@
   end
 
   describe 'Check the presence of the cloudwatch package gpg key'
-  # In Ubuntu 20.04 due to environment variable the keyring is placed under home of the user ubuntu with the permission of root
-  keyring = os_properties.ubuntu2004? && !os_properties.virtualized? ? '--keyring /home/ubuntu/.gnupg/pubring.kbx' : ''
   sudo = os_properties.redhat_ubi? ? '' : 'sudo'
+  keyring = ''
+  if os_properties.ubuntu2004?
+    # In Ubuntu 20.04 due to environment variable the keyring is placed under home of the user ubuntu with the permission of root
+    # We don't check exit status for Ubuntu20 because it returns 2 when executed in the validate phase of a created AMI
+    keyring = !os_properties.on_docker? ? '--keyring /home/ubuntu/.gnupg/pubring.kbx' : ''
+  else
+    describe bash("#{sudo} gpg --list-keys #{keyring}") do
+      its('exit_status') { should eq 0 }
+    end
+  end
   describe bash("#{sudo} gpg --list-keys #{keyring}") do
```

Comment:
It might be better to put an if clause inside the describe to avoid calling gpg list keys twice?
```
its('exit_status') { should eq 0 } unless os_properties.ubuntu2004?
```
Is this valid syntax?

Reply:
You need to declare a variable outside and use it inside.

Outside: `ubuntu2004 = os_properties.ubuntu2004?`.

Inside: `unless ubuntu2004`.

Reply:
Thanks @awslitvit this solved this issue I was facing, that was:

``` 
>>>>>>     Failed to complete #verify action: [`os_properties` is not available on an example group (e.g. a `describe` or `context` block). It is only available from within individual examples (e.g. `it` blocks) or from constructs that run in the scope of an example (e.g. `before`, `let`, etc).] on cloudwatch-ubuntu2004
```